pool/opts.go

```diff
@@ -52,6 +53,18 @@ func OnFullBuffer(size int, drainInterval time.Duration) Opt {
 	}
 }
 
+// GetTimeout effects the Pool's behavior when it is empty and create rate
```

Comment:
This comment should be reworded. The GetTimeout will affect all Get's, not just when the create rate limit is enabled. Which is better anyway

Also this comment should mention that the timeout doesn't include the time it might take to dial the connection, since that can't be canceled midway (at least not without making a lot of changes to the main package to allow passing contexts into Dial, and possibly breaking the signature of DialFunc).